src/main/java/org/mule/tools/apikit/output/MuleConfigGenerator.java

```diff
@@ -64,112 +71,249 @@ public MuleConfigGenerator(List<ApikitMainFlowContainer> apis, List<GenerationMo
+      flowEntries.forEach(flowEntry -> generateForFlowEntries(flowEntry));
     }
-    return configs;
+    return new ArrayList<>(muleConfigs);
   }
 
-  private void setDefaultApikitAndListenersConfigs(ApikitMainFlowContainer api, MuleConfig muleConfig) {
-    if (api.getConfig() == null || api.getHttpListenerConfig() == null) {
-      if (api.getConfig() == null) {
-        api.setDefaultAPIKitConfig();
-      }
-      generateAPIKitAndListenerConfig(api, muleConfig);
+  /**
+   * Looks for main mule configuration file and updates apikit configuration in case needed. Adds both files to
+   * mule configurations list if applies.
+   * @param api container of the main mule application file
+   * @return
+   */
+  private List<MuleConfig> generateBasicMuleConfiguration(ApikitMainFlowContainer api) {
+    List<MuleConfig> muleConfigs = new ArrayList<>();
+    MuleConfig mainMuleConfig = api.getMuleConfig() != null ? api.getMuleConfig() : createMuleConfig(api);
+    MuleConfig apikitConfig = updateApikitConfig(api, mainMuleConfig);
+    muleConfigs.add(mainMuleConfig);
+    if (apikitConfig != null) {
+      muleConfigs.add(apikitConfig);
     }
+    return muleConfigs;
+  }
+
+  /**
+   * Generates new flows based on the generation models previously created. Also updates apikit configuration in case
+   * needed.
+   * @param flowEntry new flow to be generated
+   */
+  private void generateForFlowEntries(GenerationModel flowEntry) {
+    Element apikitFlowScope = new APIKitFlowScope(flowEntry, isMuleEE()).generate();
+    MuleConfig mainMuleConfig = flowEntry.getApi().getMuleConfig();
+    mainMuleConfig.getContentAsDocument().getRootElement().getContent().add(apikitFlowScope);
+    mainMuleConfig.addFlow(new Flow(apikitFlowScope));
   }
 
-  private int getLastFlowIndex(Document doc) {
-    int lastFlowIndex = 0;
-    for (int i = 0; i < doc.getRootElement().getContentSize(); i++) {
-      Content content = doc.getRootElement().getContent(i);
-      if (content instanceof Element && "flow".equals(((Element) content).getName())) {
-        lastFlowIndex = i;
-      }
+  /**
+   * It looks for the first (and only) apikit router configuration reference within the mainMuleConfig
+   * received by parameter. Then with that it looks inside all of the files of the app for the file which
+   * calls that reference.
+   * @param mainMuleConfig the configuration which contains the reference to the apikit configuration.
+   * @return the file on which apikit configuration is made
+   */
+  private MuleConfig retrieveApikitConfigMuleConfig(MuleConfig mainMuleConfig) {
+    if (muleConfigsInApp.size() > 1) {
+      String configRef = mainMuleConfig.getMainFlows().stream().findFirst().get().getApikitRouter().getConfigRef();
+      Stream<MuleConfig> apikitConfigurations = muleConfigsInApp.stream().filter(muleConfig -> muleConfig.getContentAsDocument()
+          .getRootElement().getChild("config", APIKitTools.API_KIT_NAMESPACE.getNamespace()) != null);
```

Comment:
This function is defined in https://github.com/mulesoft/apikit-rest-scaffolder/pull/85/files#diff-239730ce83f05a28f3dddc091bb60fdfR178